tests/modules/webp-uploads/webp-uploads-test.php

```diff
@@ -0,0 +1,62 @@
+<?php
+/**
+ * Tests for webp-uploads module.
+ *
+ * @package performance-lab
+ */
+
+class WebP_Uploads_Tests extends WP_UnitTestCase {
+
+	/**
+	 * Test if webp-uploads applies filter based on system support of WebP.
+	 */
+	function test_webp_uploads_applies_filter_based_on_system_support() {
+		// Set the expected value based on the underlying system support for WebP.
+		if ( ! wp_image_editor_supports( array( 'mime_type' => 'image/webp' ) ) ) {
+			$expect = array();
+		} else {
+			$expect = array( 'image/jpeg' => 'image/webp' );
+		}
+
+		$output_format = apply_filters( 'image_editor_output_format', array(), '', 'image/jpeg' );
+		$this->assertEquals( $expect, $output_format );
+	}
+
+	/**
+	 * Verify webp-uploads applies filter with a system that supports WebP.
+	 */
+	function test_webp_uploads_applies_filter_when_system_supports() {
```

Comment:
Based on my comment below:
```suggestion
	function test_webp_uploads_filter_image_editor_output_format_with_support() {
```